### PR TITLE
Smooch - Add event definitions

### PR DESCRIPTION
**Pull request by drawm**
- Changes are based on [smooch examples](https://github.com/zendesk/sunshine-conversations-web/tree/2e2ac641d34bb6762673750b65773e17739a4aa1#events)
- Move interfaces to global.d.ts to expose them without breaking the current way Smooch is imported
- Add events method definition to smooch types
- Add tests for smooch events

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Smooch event examples](https://github.com/zendesk/sunshine-conversations-web/tree/2e2ac641d34bb6762673750b65773e17739a4aa1#events)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/smooch/index.d.ts [hunk 1]
```diff
@@ -180,769 +180,87 @@ declare namespace Smooch {
...
-
-    interface Status {
-        status: 'resolved' | 'pending' | string;
-    }
-
-    interface ConversationData {
-        conversation: Conversation;
-    }
-
+    // tslint:disable:unified-signatures
```
Comment (drawm) [inline types/smooch/index.d.ts:183]: Each definition can change independently in the future, therefore its best to keep them separated. 

It also helps with comments and readability.

**Review by jpsorensen — APPROVED** [review]
@drawm congrats and thanks for your first contribution!

I'm new to the repo myself, and I wasn't sure whether inserting interface definitions into the global namespace would be discouraged (especially with generic names like Status and User).

However, poking around a bit I do see examples of other type definitions doing this (for example, the React definitions insert empty definitions of several DOM-related types), and the following line from the FAQ makes me think it's allowed:

> I'm writing a definition that depends on another definition. Should I use <reference types="" /> or an import?
> 
> If the module you're referencing is an external module (uses export), use an import. If the module you're referencing is an ambient module (uses declare module, or just declares globals), use \<reference types="" />.

I'm curious how the unified-signatures rule could allow you to express that a value in one parameter implies a specific type in another param. I like how your definitions give the developer more information - if my event is 'ready', my callback will take no params.

Looks great, thanks!

**Comment by drawm** [comment]
Ready to merge

**Comment by drawm** [comment]
@jpsorensen Glad you like the changes :) 

> I'm curious how the unified-signatures rule could allow you to express that a value in one parameter implies a specific type in another param.

IIRC `unified-signatures ` it uses union type (`|`) to group the changes and leave the similar types alone. Its not implying that a type is in used based on another type. 

It might be possible to do so using generics + [conditional types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html) but I doubt it will be as clean a grouping similarities with union group. Unless the methods you are trying to type are dynamically created, its better to simply define all the possibilities and group those you know won't change using `unified-signatures` to save space.

**Pull request merged**